app/src/main/java/com/android/streetworkapp/ui/navigation/InfoDialogs.kt

```diff
@@ -0,0 +1,137 @@
+          screens.EVENT_OVERVIEW to
+              InfoDialog(
+                  "EventOverview",
+                  "What is the Event Overview ?",
+                  "This is the event overview, here you can access all of the information that you need. \n\n You are free to join the event if you want to participate !"),
+          screens.PROGRESSION to
+              InfoDialog(
+                  "Progression",
+                  "What is the Progression ?",
+                  "This is the progression screen, here you can see your progression and achievements. \n\n There is your record for each exercise in Training, click on it to see more details !")
+
+          // Auth not done because unnecessary
+          // Add more if needed
+          )
+
+  /** Returns the default InfoDialog (for screens that are not supported). */
+  fun defaultInfoDialog(): InfoDialog {
+    return InfoDialog(
+        "Default",
+        "Screen not supported yet",
+        "Information about this screen is not available. Please explore to learn more!")
+  }
+
+  /**
+   * This function sets up the INFO action callback for the dialog. Cannot be done in another way
+   * because it needs the topAppBarManager to be instantiated.
+   */
+  fun setUp() {
+    Log.d("InfoDialog", "Setting up actions callbacks with Manager $topAppBarManager")
+    topAppBarManager!!.setActionCallback(TopAppBarManager.TopAppBarAction.INFO) {
+      Log.d("InfoDialog", "Info callback start : show = ${showDialog.value}")
+      showDialog.value = true
+      Log.d("InfoDialog", "Info callback end : show = ${showDialog.value}")
+    }
+  }
+
+  /**
+   * This function displays the info dialog for the current screen. If the screen is not supported,
+   * a default "non-supported" version is displayed.
+   */
+  @Composable
+  fun Display() {
+    Log.d("InfoDialog", "Displaying info dialog ${showDialog.value} for screen ${screenName.value}")
+    val func = infoDialogs[screenName.value]
+    if (func != null) {
+      Log.d("InfoDialog", "Displaying info dialog : ${func.title}")
+      func.DisplayInfoDialog(showDialog)
+    } else {
+      Log.d("InfoDialog", "Displaying default info dialog : ${defaultInfoDialog().title}")
+      defaultInfoDialog().DisplayInfoDialog(showDialog)
+    }
+  }
+
+  /** Returns the InfoDialogs mapping */
+  fun getInfoDialogs(): MutableMap<String, InfoDialog> {
+    return infoDialogs
+  }
+}
+
+class InfoDialog(val tag: String, val title: String, val content: String) {
+
+  /**
+   * This function displays the info dialog, which is a specific Type of CustomDialog.
+   *
+   * @param showDialog - The display state of the dialog.
+   */
+  @Composable
+  fun DisplayInfoDialog(showDialog: MutableState<Boolean>) {
+    CustomDialog(
+        showDialog,
+        dialogType = DialogType.INFO,
+        tag + "Info",
```

Comment:
I personally find `"${tag}Info"` cleaner, but that's just a suggestion, fell free to leave it _and the all the others I have commented_ like that.

Reply:
I agree, I will change it right now.